<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html>
<head>
  <meta name="generator" content=
  "HTML Tidy for Mac OS X (vers 31 October 2006 - Apple Inc. build 15.6), see www.w3.org">

  <title>Manager documentation</title>
  <style type="text/css">
    body {
        font-size: 80%;
        font-family: "Lucida Grande", Verdana, Arial, Sans-Serif;
      }
      h2 {
        border-bottom: solid 1px black;
      }
      .description {
        background: #fcc;
        border: solid 1px #800;
        padding: 1em;
      }
      .method {
        background: #ccc;
        border: solid 1px #444;
        padding: 0.5em 1em 0.5em 1em;
        margin: 1em;
      }
      .back {
        text-align: right;
      }
  </style>
</head>

<body>
  <a name='top' id="top"></a>

  <ul>
    <li><a href='#balloon'>balloon</a></li>

    <li><a href='#bioclipse'>bioclipse</a></li>

    <li><a href='#biojava'>biojava</a></li>

    <li><a href='#biows'>biows</a></li>

    <li><a href='#browser'>browser</a></li>

    <li><a href='#cdk'>cdk</a></li>

    <li><a href='#cdx'>cdx</a></li>

    <li><a href='#chemspider'>chemspider</a></li>

    <li><a href='#ds'>ds</a></li>

    <li><a href='#gist'>gist</a></li>

    <li><a href='#hivdrc'>hivdrc</a></li>

    <li><a href='#inchi'>inchi</a></li>

    <li><a href='#jcp'>jcp</a></li>

    <li><a href='#jcpglobal'>jcpglobal</a></li>

    <li><a href='#jmol'>jmol</a></li>

    <li><a href='#js'>js</a></li>

    <li><a href='#kalignws'>kalignws</a></li>

    <li><a href='#matrix'>matrix</a></li>

    <li><a href='#metaprint2d'>metaprint2d</a></li>

    <li><a href='#myexperiment'>myexperiment</a></li>

    <li><a href='#opentox'>opentox</a></li>

    <li><a href='#opsin'>opsin</a></li>

    <li><a href='#owl'>owl</a></li>

    <li><a href='#pellet'>pellet</a></li>

    <li><a href='#qsar'>qsar</a></li>

    <li><a href='#r'>r</a></li>

    <li><a href='#rdf'>rdf</a></li>

    <li><a href='#sadi'>sadi</a></li>

    <li><a href='#signatures'>signatures</a></li>

    <li><a href='#twitter'>twitter</a></li>

    <li><a href='#ui'>ui</a></li>

    <li><a href='#userManager'>userManager</a></li>

    <li><a href='#ws'>ws</a></li>

    <li><a href='#xml'>xml</a></li>

    <li><a href='#xmpp'>xmpp</a></li>
  </ul>

  <h2 id='balloon'>balloon</h2>

  <p class='description'>Contains methods related to Balloon</p>

  <div class='method'>
    <h3>generateMultiple3Dconformations(List&lt;IMolecule&gt;
    molecules, int numConf)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Return a list of molecule with the selected number of
      target 3D conformations generated by Balloon.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dconformations(String inputfile, String
    outputfile, int numConformations)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generated 3D conformations for all molecules in the
      inputfile.<br>
      Inputs: inputfile = path to inputfile, outputfile = path to
      outputfile, numConformations = number of conformations to
      generate per molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dconformations(IMolecule molecule, int
    numConf)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Return a list of molecule with selected number of target
      3D conformations generated by Balloon.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateMultiple3Dcoordinates(List&lt;IMolecule&gt;
    molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Return a list of molecules with 3D coordinates generated
      by Balloon.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dcoordinates(String inputfile)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generated 3D coodrinates for all molecules in the
      inputfile. Creates a default output file. Inputs: inputfile =
      path to inputfile.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dcoordinates(String inputfile, String
    outputfile)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generated 3D coodrinates for all molecules in the
      inputfile and saves to outputfile Inputs: inputfile = path to
      inputfile, outputfile = path to outputfile</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dcoordinates(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Return a new molecule with 3D coordinates generated by
      Balloon.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dcoordinates(List&lt;String&gt; inputfiles)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate 3D coordinates for all molecules in a list of
      inputfiles. Creates a default output file. Inputs: inputfiles
      = list of Strings with paths to inputfiles</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dconformations(String inputfile, int
    numConformations)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate 3D conformations for all molecules in an
      inputfile. Creates a default output filename. inputfile =
      paths to inputfile, numConformations = number of
      conformations to generate per molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3Dconformations(List&lt;String&gt; inputfiles, int
    numConformations)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate 3D conformations for all molecules in a list of
      inputfiles. inputfiles = list of Strings with paths to
      inputfiles, numConformations = number of conformations to
      generate per molecule.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='bioclipse'>bioclipse</h2>

  <p class='description'>The Bioclipse Platform manager is used for
  providing some textual information on the Bioclipse platform.</p>

  <div class='method'>
    <h3>downloadAsFile(String url, String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download the URL content into a file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>download(String url)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download the URL content as String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>bugTracker()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens the Bioclipse Bug Tracker.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>planet()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens Planet Bioclipse.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>assumeOnline()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Throws an exception if Bioclipse does not have internet
      access.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isOnline()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if Bioclipse is connected to the
      internet.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>requireVersion(String lowerVersionBound, String
    upperVersionBound )</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the current Bioclipse version is in
      between the given (including) lower version bound and the
      given (excluding) upper version bound<br>
      So bioclipse.requireVersion will</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadAsFile(String url, String mimeType, String
    filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download the URL content into a file while passing the
      given MIME type as Accept: HTTP header.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>version()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a string representation of the current Bioclipse
      version</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>download(String url, String mimeType)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download the URL content as String while passing the
      given MIME type as Accept: HTTP header.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>wiki()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens the Bioclipse Wiki.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>requireVersion(String version)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the given version is lower than the
      current Bioclipse version</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='biojava'>biojava</h2>

  <p class='description'>Provides bioinformatics services through
  the BioJava project.</p>

  <div class='method'>
    <h3>RNAfromPlainSequence(String rnaString, String name)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaRNA sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>RNAfromPlainSequence(String rnaString)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaRNA sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAtoRNA(IDNA dna)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the RNA sequence equivalent of the given DNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>proteinsFromString(String proteinString)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaProtein sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>RNAtoProtein(List&lt;IRNA&gt; rnas)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalents of the given
      list of RNA sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAtoRNA(IDNA dna, String name))</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the RNA sequence equivalent of the given DNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>translationOf(IRNA rna, String name))</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalent of the given RNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>translationOf(List&lt;IRNA&gt; rnas)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalents of the given
      list of RNA sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createSequenceList()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create an empty list of Sequences</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>proteinsToFASTAfile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves an array of proteins to a FASTA file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAfromPlainSequence(String dnaString)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaDNA sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>translationOf(IRNA rna)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalent of the given RNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>proteinFromPlainSequence(String proteinString)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaProtein sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>RNAtoProtein(IRNA rna)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalent of the given RNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>transcriptionOf(IDNA dna)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the RNA sequence equivalent of the given DNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>sequencesToFASTAfile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves an array of sequences to a FASTA file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>proteinsFromFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns an array of proteins from the given file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAfromPlainSequence(String dnaString, String name)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaDNA sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>RNAtoProtein(IRNA rna, String name))</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalent of the given RNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>proteinFromPlainSequence(String proteinString, String
    name)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaProtein sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>features(ISequence sequence)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Return a list of features for a Sequence</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAtoProtein(List&lt;IDNA&gt; dnas)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of protein sequence equivalents of the
      given list of DNA sequences.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>RNAsFromString(String rnaString)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaRNA sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>sequencesFromString(String s)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates sequences from a String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>transcriptionOf(IDNA dna, String name))</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the RNA sequence equivalent of the given DNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAsFromFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns an array of DNA sequences from the given
      file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>sequencesToFASTAString(List&lt;ISequence&gt;
    sequences)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Serialize a list of sequences to a FASTA String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>transcriptionOf(List&lt;IDNA&gt; dnas)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of RNA sequence equivalents of the given
      list of DNA sequences.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAsFromString(String dnaString)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a new BiojavaDNA sequence from the given
      String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>RNAsFromFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns an array of RNA sequences from the given
      file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAtoRNA(List&lt;IDNA&gt; dnas)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of RNA sequence equivalents of the given
      list of DNA sequences.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAtoProtein(IDNA dna, String name))</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalent of the given DNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>dnaToFASTAString(List&lt;IDNA&gt; dnas)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Serialize a list of DNA sequences to a FASTA String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>sequencesFromFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads sequences from file at path</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>DNAtoProtein(IDNA dna)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the protein sequence equivalent of the given DNA
      sequence.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>proteinsToFASTAString(List&lt;IProtein&gt; proteins)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Serialize a list of protein sequences to a FASTA
      String.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='biows'>biows</h2>

  <p class='description'>Controls access to Bioinformatics
  Webservices.</p>

  <div class='method'>
    <h3>queryUniProtKB(String seqid)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download protein sequences in FASTA format from the
      Uniprot KnowwledgeBase at EBI. Entries can be retrieved by
      entry name (e.g. INSR_HUMAN) or by accession number (e.g.
      P06213).</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>queryRefseq(String seqid)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download DNA sequences in FASTA format from the refseq
      database at EBI. Seqid should be RefSeq ID separated by comma
      OR space, e.g. NM_</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>queryEMBL(String seqid)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download DNA sequences in FASTA format from the EMBL
      nucleotide sequence database at EBI. Seqid can be either
      accession number (e.g. M10051) or entry name (e.g.
      HSINSR).</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='browser'>browser</h2>

  <p class='description'>A Manager for scraping web pages</p>

  <div class='method'>
    <h3>scrapeWebpage(String url)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Scrapes a web URL by content and returns a List of
      IBioObjects</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='cdk'>cdk</h2>

  <p class='description'>Provides CDK-derived functionality.</p>

  <div class='method'>
    <h3>determineIChemFormatOfString(String fileContent)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Determines the file format of the file in the String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>randomSplit(List&lt;IMolecule&gt; mols, int parts)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Split a list of molecules in N parts.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setProperty(ICDKMolecule molecule, Object propertyName,
    Object propertyValue)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the property value for the given property name.
      Returns the old value, or null if it was yet unset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getFormats()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns all available IChemFormats</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>asList(IMoleculeSet set)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Converts a CDK IMoleculeSet object into a
      List&lt;ICDKMolecule&gt;.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>clone(ICDKMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Clones the CDK data model.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isConnected(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the given atom container is fully
      connected. Use partition directly if that is the antipicated
      next action, to no repeat to same computation.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>determineIChemFormat(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Determines the file format of the file, if chemical</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>fingerPrintMatches(ICDKMolecule molecule, ICDKMolecule
    subStructure)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the fingerprint of the subStructure is a
      subset of the fingerprint for the molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addExplicitHydrogens(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds explicit hydrogens to this molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol, boolean overwrite)</h3>

    <dl>
      <dt>summary</dt>

      <dd>saves mol to a file, if previously read from file.
      Overwrite determines if existing file shall be
      overwritten.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol, String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>saves mol to a file (filename must be a relative to
      workspace root and folder must exist), filetype must be one
      of the constants given by getPossibleFiletypes</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>removeImplicitHydrogens(ICDKMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Removes implicit hydrogens</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getInfo(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns information about molecules in file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>newMolecule()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a cdk molecule from SMILES</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>extractFromSDFile(String file, int startentry, int
    endentry)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Extracts a number of entries from an sd file. Does not
      read the complete file for this.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>perceiveAromaticity(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Perceives aromaticity on an IMolecule. The molecule will
      be unchanged and a new molecule created. Uses the
      org.openscience.cdk.aromaticity.CDKHueckelAromaticityDetector
      class from CDK.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createMoleculeList()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates an empty molecule list.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createSDFileIndex(String filePath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a index file for the SD-file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>fromString(String input file)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a cdk molecule from a String. Guessing the format
      of the String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createConformerIterator(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns an iterator to the molecules in an IFile that
      might contain conformers.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMDLMolfile(ICDKMolecule molecule, String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves a molecule in the MDL molfile V2000 format
      (filename must be a relative to workspace root and folder
      must exist)</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>asCDKMolecule(IMolecule m)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a cdk molecule from a molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol, String filename, IChemFormat
    filetype, boolean overwrite, Properties writerProperties)</h3>

    <dl>
      <dt>summary</dt>

      <dd>saves mol to a file (filename must be a relative to
      workspace root and folder must exist), filetype must be a
      IChemFormat. If overwrite=true then file will be overwritten
      if exists. Properties can be given to tune the output of the
      writer.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>smartsMatches(ICDKMolecule molecule, String smarts)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the given SMARTS matches the given
      molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>has2d(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if molecule has 2D coordinates, false
      otherwise.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>moleculesFromString(String s)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a list of molecules from a given String, guessing
      the format of the String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>asSet(List&lt;ICDKMolecule&gt; list)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Converts a Bioclipse List&lt;ICDKMolecule&gt; list into a
      CDK IMoleculeSet.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>randomSplit2parts(List&lt;IMolecule&gt; mols, double
    firstRatio)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Split a list of molecules in 2 parts by a ratio.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getNoMolecules(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns number of molecules in file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateMass(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate and return the molecular weight for the
      molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>partition(IMolecule molecule: atom container to
    fragmentate)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Splits up an atom container into a List of
      IAtomContainer"s with fully connected molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>removeExplicitHydrogens(ICDKMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Removes explicit hydrogens</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>mcss(molecules List of molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Determines the Maximal Common SubStructure (MCSS) for the
      list of molecules.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateRMSD(List&lt;IMolecule&gt; molecules, String
    filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate similarities between all molecules in the list
      based on the RMSD between the the Kabsch-aligned MCSSs.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveCML(ICDKMolecule molecule, String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves a molecule in the Chemical Markup Language format
      (filename must be relative to workspace root and folder must
      exist). Example of file String: \"/Virtual/bla.cml\"</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>molecularFormula(ICDKMolecule m)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Gives the molecularformula as a String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>has3d(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if molecule has 3D coordinates, false
      otherwise.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>appendToSDF(String sdFile, ICDKMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Append a molecule at the end of a given sd file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateTanimoto(BitSet fingerprint1, BitSet
    fingerprint2)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate tanimoto similarity between two
      fingerprints</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>numberOfEntriesInSDF(String filePath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Counts the number of entries in an SDF file at the given
      file path. Returns 0 in case of problem.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>subStructureMatches(ICDKMolecule molecule, ICDKMolecule
    subStructure)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the paramater named subStructure is a
      substructure of the paramater named molecule.<br>
      (Performs an isomophism test without checking
      fingerprints)</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>kabsch(molecules List of molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Aligns the molecular geometries to the first molecule
      based on their MCSS.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>determineFormat(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Determines the file format if the file, if chemical</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>loadConformers(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads the molecules at the path into a list, and store
      conformers in a single molecule. Currently only reads
      SDFiles.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>newMolecule(IAtomContainer atomContainer)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a Bioclipse IMolecule object from a CDK
      IAtomContainer.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>fromCml(String cml)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a cdk molecule from a CML String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>areIsomorphic(ICDKMolecule molecule1, ICDKMolecule
    molecule2)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the paramater named molecule1 and the
      paramater named molecule2 are isomorph.<br>
      (Performs an isomophism test without checking
      fingerprints)</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol, String filename, IChemFormat
    filetype, boolean overwrite)</h3>

    <dl>
      <dt>summary</dt>

      <dd>saves mol to a file (filename must be a relative to
      workspace root and folder must exist), filetype must be a
      IChemFormat. If overwrite=true then file will be overwritten
      if exists.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol, String filename, IChemFormat
    filetype)</h3>

    <dl>
      <dt>summary</dt>

      <dd>saves mol to a file (filename must be a relative to
      workspace root and folder must exist), filetype must be a
      IChemFormat</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateSMILES(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the SMILES for a molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>extractFromSDFile(String file, String property, Collection
    values)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Extracts moleculs indicated by index in values to a
      list</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addImplicitHydrogens(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds implicit hydrogens to this molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3dCoordinates(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create 3D coordinates for the given molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getFormat(String type)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the IChemFormat for the given type</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getTautomers(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generates a list of tautomers.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createSDFPropertyMap(String file, String property)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create a index for molecules that contains a certain
      property</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getSmartsMatches(ICDKMolecule molecule, String SMARTS)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query a molecule for a SMARTS string and return a list of
      IAtomContainers with the matches.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>determineFormat(IContentType type)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Determines the IChemFormat equivalent of the given
      content type</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate2dCoordinates(List&lt;IMolecule&gt; molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create 2D coordinates for the given molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getMDLMolfileString(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a MDL V2000 molfile serialization</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>guessFormatFromExtension(String type)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Makes an educated guess if the file format based on the
      file extension alone and returns a CDKIChemFormat
      object.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>loadSMILESFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads molecules from a SMILES file at a given path into a
      list of molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves molecule to file, if read from file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveMolecule(IMolecule mol, String filename, boolean
    overwrite)</h3>

    <dl>
      <dt>summary</dt>

      <dd>saves mol to a file (filename must be a relative to
      workspace root and folder must exist), with a given filename
      and overwrites determined by the given boolean. File type is
      taken from the mol object, if available. If not, then the
      file extension is used to make a somewhat educated
      guess.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate3dCoordinates(List&lt;IMolecule&gt; molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create 3D coordinates for the given molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateTanimoto(List&lt;IMolecule&gt; molecules, String
    filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate tanimoto similarities between all molecule in
      the list using CDK fingerprints.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>subStructureMatches(List&lt;ICDKMolecule&gt; molecules,
    ICDKMolecule subStructure)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns all molecules in the given list which have a
      given subStructure as subgraph.<br>
      (Performs an isomophism test without checking
      fingerprints)</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateTanimoto(IMolecule calculateFor, IMolecule
    reference)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate tanimoto similarity of calculateFor and
      reference via CDK fingerprint.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getSubstructures(ICDKMolecule molecule, ICDKMolecule
    subStructure)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the substructure matches in the molecule for the
      given substructure</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateTanimoto(BioList&lt;IMolecule&gt; calculateFor,
    BitSet reference)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate tanimoto similarity of a molecule
      (calculateFor) to a BitSet (reference) via CDK
      fingerprint.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate2dCoordinates(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create 2D coordinates for the given molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isValidSmarts(String SMARTS)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Determines if a SMARTS string can be interpreted by
      CDK.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>loadMolecule(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads a molecule from file. Does not do aromaticity
      detection (use perceiveAromaticity). Returns the first if
      multiple molecules exists in the file</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>fromSMILES(String SMILES)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a cdk molecule from SMILES</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createMoleculeIterator(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates and iterator to the molecules in the file at the
      path</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateTanimoto(BioList&lt;IMolecule&gt; calculateFor,
    IMolecule reference)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate tanimoto similarity of a list of molecules
      (calculateFor) to another molecule (reference) via CDK
      fingerprint.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>loadMolecules(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads molecules from a file at a given path into a list
      of molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getProperty(ICDKMolecule molecule, Object
    propertyName)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the property value for the given property
      name</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveSDFile(String file, List&lt;IMolecule&gt; entries)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates an sd file from a number of molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>totalFormalCharge(IMolecule molecule: molecule to calculate
    the total formal charge for.)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculates the total formal charge.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='cdx'>cdx</h2>

  <p class='description'>Contains CDK debug related methods</p>

  <div class='method'>
    <h3>listWriterOptions(IChemFormat format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a writer options for the CDK writer for the given
      IChemFormat.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>perceiveSybylAtomTypes(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of Sybyl atom types.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>perceiveCDKAtomTypes(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of CDK atom types.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>debug(ICDKMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a string representation of the data
      structures.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listReaderOptions(IChemFormat format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a reader options for the CDK reader for the given
      IChemFormat.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>diff(ICDKMolecule mol, ICDKMolecule mol2)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the differences between the two molecules</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='chemspider'>chemspider</h2>

  <p class='description'>Manager that allows interaction with the
  ChemSpider database.</p>

  <div class='method'>
    <h3>substructureSearch(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query ChemSpider for compounds where input is a
      substructure</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>download(Integer csid)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads the ChemSpider compound with the given identifier
      into a IMolecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>simpleSearch(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query ChemSpider for a molecular structure</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>resolve(String inchiKey)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a List of matching compound CSIDs.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>loadCompound(int csid, String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads the ChemSpider MDL molfile with the given
      ChemSpider ID (csid) to the given path</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>exactSearch(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query ChemSpider for exact matches</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadAsString(Integer csid)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads the ChemSpider compound with the given identifier
      into a String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>similaritySearch(IMolecule molecule, Float tanimoto)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query ChemSpider for similar molecules within a certain
      tanimoto distance.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='ds'>ds</h2>

  <p class='description'>Contains methods for Bioclipse Decision
  Support</p>

  <div class='method'>
    <h3>runTest(String testID, IMolecule mol,)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run a Decision Support Test on a Molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getEndpoints()</h3>

    <dl>
      <dt>summary</dt>

      <dd>List available Decision Support Endpoints</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getTests()</h3>

    <dl>
      <dt>summary</dt>

      <dd>List available Decision Support Tests</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>runEndpoint(String endpointID, IMolecule mol,)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run a Decision Support Test for an endpoint</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='gist'>gist</h2>

  <p class='description'>The gist manager is used for downloading
  gists</p>

  <div class='method'>
    <h3>download(int gist)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Downloads the Gist with the given number to the project
      "Gists/" and returns the path</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='hivdrc'>hivdrc</h2>

  <p class='description'>A manager for invoking the HIVDRC
  services.</p>

  <div class='method'>
    <h3>predictPI(String sequence, String JID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Invoke the hivpred XMPP service on a list of protease
      sequences, each comprising a 99 aa long string in one letter
      aa code, on a specified JID of an existing XMPP hivpred
      service</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>predictPI(String sequence)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Invoke the hivpred XMPP service on a list of protease
      sequences, each comprising a 99 aa long string in one letter
      aa code.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>predictNRTI(String sequence, String JID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Invoke the hivpred XMPP service on a list of protease
      sequences, each comprising a 99 aa long string in one letter
      aa code, on a specified JID of an existing XMPP hivpred
      service</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>predictNRTI(String sequence)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Invoke the hivpred XMPP service on a list of NRTI
      sequences, each comprising a 240 aa long string in one letter
      aa code.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='inchi'>inchi</h2>

  <p class='description'>Manager for creating InChI and
  InChIKeys.</p>

  <div class='method'>
    <h3>generate(IMolecule molecule, String options)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generates the InChI and InChIKey for the given molecule,
      using the given options. This options String consists of one
      or more, space-delimited options, such as FixedH.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generates the InChI and InChIKey for the given
      molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isAvailable()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the InChI library is available for
      yourplatform.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>options()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of InChI generation options.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isLoaded()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if the InChI library could be loaded.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>load()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads the InChI library.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='jcp'>jcp</h2>

  <p class='description'>Manager for the JChemPaintEditor scripting
  language.</p>

  <div class='method'>
    <h3>addPhenyl(IBond bond)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a phenyl ring fused with the given bond.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addRing(IBond bond, int size)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a carbon ring of the given size fused with the given
      bond.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>updateImplicitHydrogenCounts()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Updates the implicit hydrogen counts, given the current
      connectivity.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>updateView()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Refreshes the JChemPaint view screen.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getIsCompact()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether atoms are compact.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setZoom(double zoom)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the zoom factor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setMassNumber(IAtom atom, int charge)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the mass number of this element.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setOrder(IBond bond, IBond.Order order)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the order of the bond.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setHighlightBondDistance(double highlightDistance)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the distance within which highlighting occurs for
      bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getZoom()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the zoom factor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>snapshot(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Takes a snapshot.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setSymbol(IAtom atom, String symbol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the element of this atom to the given
      symbol.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setHighlightAtomDistance(double highlightDistance)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the distance within which highlighting occurs for
      atoms.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getModel()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the ICDKMolecule of theactive JChemPaint
      editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>zap()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Deletes all atoms and bonds in the model.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addAtom(String elementSymbol, IAtom atom)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a new atom bonded to the given atom.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setWedgeType(IBond bond, int type)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the wedge type of the bond.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addBond(IAtom atom, int charge)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create a new bond between the two given atoms.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setDrawNumbers(boolean setDrawNumbers)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether atom numbers are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setBondDistance(double bondDistance)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets distance between lines in multiple bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDrawNumbers()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether atom numbers are showed.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowEndCarbons(boolean showEndCarbons)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether explicit methyl groups at the end of chains
      are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setRingProportion(double ringProportion)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the position of inner-ring bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowAromaticityInCDKStyle()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether CDK-style aromiticity is used.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getFitToScreen()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether the diagram fits the screen.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getScale()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Gets the scale paramterer.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setModel()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the ICDKMolecule of the active JChemPaint
      editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addRing(IAtom atom, int size)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a carbon ring of the given size to the given
      atom.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowEndCarbons()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether explicit methyl groups are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowAromaticity()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether aromatic indicators are on.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setMargin(double margin)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the margin of the diagram.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>moveTo(IBond bond, Point2d point)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Moves the center of the bond to the new point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setBondWidth(double bondWidth)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the width of bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setIsCompact(boolean isCompact)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether atoms are drawn as compact.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setCharge(IAtom atom, int charge)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the formal charge of this atom.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getRingProportion()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the position of inner-ring bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setFitToScreen(boolean fitToScreen)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Resizes the diagram to fit the screen.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>newPoint2d(double x, double y)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new javax.vecmath.Point2d.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addExplicitHydrogens()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds explicit hydrogens, given the current
      connectivity.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondLength()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the standard bond length.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>selectPart(IChemObject element)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Selectes all atoms and bonds that is connected to the
      selected element</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowImplicitHydrogens(boolean implicitHydrogens)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether implicit hydrogens are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getHighlightAtomDistance()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the distance to within which highlighting occurs
      for atoms.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>selectAll()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Selectes all atoms and bonds</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getHighlightBondDistance()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the distance to within which highlighting occurs
      for bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addAtom(String elementSymbol, Point2d worldcoord)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a new atom at the given coordinates.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowExplicitHydrogens()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether explicit hydrogens are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setAtomRadius(double atomRadius)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the radius of an atom symbol.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondWidth()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the width of bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getWedgeWidth()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the on-screen width of a wedge.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>removeAtom(IAtom atom)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Removes an IAtom from the data model.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setScale(double new scale)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets a new scale</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getAtomRadius()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the radius of an atom symbol.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setWedgeWidth(double wedgeWidth)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the on-screen width of a wedge.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getClosestAtom(Point2d atomRadius)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the IAtom closest to the given world
      coordinate.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>moveTo(IAtom atom, Point2d point)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Moves an atom to the given location.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondDistance()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the distance between multiple bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addPhenyl(IAtom atom)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a phenyl ring to the given atom.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>removeExplicitHydrogens()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Removes all explicit hydrogens</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getClosestBond(Point2d worldCoord)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the IBond closest to the given world
      coordinate.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowImplicitHydrogens()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether implicit hydrogens are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondOrder(int order)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns an instance of IBond.Order matching the given
      order.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>removeBond(IBond bond)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Removes an IBond from the data model.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>cleanup()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Recalculates 2D coordinates for the complete
      molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowExplicitHydrogens(boolean explicitHydrogens)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether explicit hydrogens are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowAromaticity(boolean showAromaticity)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether aromatic indicators are on.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getMargin()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the margin of the diagram.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setBondLength(double bondLength)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the standard bond length.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowAromaticityInCDKStyle(boolean
    showAromaticityCDK)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether aromaticity are shown in CDK style.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='jcpglobal'>jcpglobal</h2>

  <p class='description'>Manager for the JChemPaintEditor
  Properties scripting language.</p>

  <div class='method'>
    <h3>getAtomRadius()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the radius of a compact atom</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setBondLength(souble bondLength)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the length of a standard bond</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setWedgeWidth(double wedgeWidth)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the width on screen of a wedge</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setBondDistance(double bondDistance)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the distance between double bonds</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>applyGlobalProperties()</h3>

    <dl>
      <dt>summary</dt>

      <dd>applies the global properties to all opened JChemPaint
      editors</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondLength()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the length of a standard bond</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowEndCarbons(boolean showEndCarbons)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether explicit methyl groups are shown</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondDistance()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the length of a standard bond</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setHighlightBondDistance( double highlightDistance)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the distance within which a bond is highlighted</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getHighlightAtomDistance()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the distance within which a bond is
      highlighted</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getHighlightBondDistance()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the distance within which an atom is
      highlighted</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowExplicitHydrogens()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether explicit hydrogens are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowImplicitHydrogens(boolean
    showImplicitHydrogens)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether implicit hydrogens are shown</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowExplicitHydrogens(boolean
    showExplicitHydrogens)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether explicit hydrogens are shown</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowImplicitHydrogens()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether implicit hydrogensare shown</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowNumbers(boolean showNumbers)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether numbers on atoms are shown.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowEndCarbons()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether explicit methyl groups are shown</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setHighlightAtomDistance( double highlightDistance)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the distance within which a atom is highlighted</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowAromaticity()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether aromatic indicators are on</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setMargin(double margin)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the amount of whitespace aroundthe diagram</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setAtomRadius(double atomRadius)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the radius of a compact atom</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getBondWidth()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the width of bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setBondWidth(double bondWidth)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the width of bonds.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setShowAromaticity(boolean showAromaticity)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets whether aromatic indicators are on</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getShowNumbers()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether numbers on atoms are shown</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getMargin()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the margin size</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getWedgeWidth()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the width of a stereo wedge</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='jmol'>jmol</h2>

  <p class='description'>This manager contains methods for
  controlling jmol</p>

  <div class='method'>
    <h3>spinOff()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Causes active jmol to stop spinning molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>run(String script)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Execute the given String as a script in the active Jmol
      editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>print(String s)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Prints the given string to the Jmol console</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>snapshot(String filepath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Export as image to path</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>append(Path to file)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Append a file to the open editor, effectively opening
      multiple files on top of eachother</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>append(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Append an IMolecule to the open editor, effectively
      opening multiple files on top of each other</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getMolecules()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a List&lt;IMolecule&gt; from the current Jmol
      editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>spinOn()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Causes active jmol to spin molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>load(IMolecule file)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads an IMolecule into a new Jmol editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>load(Path to file)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Load a file into the active Jmol editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>minimize()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Optimizes the geometry of the structure in the active
      JmolEditor (but not back to file)</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='js'>js</h2>

  <p class='description'>Controls access to the JavaScript
  Console.</p>

  <div class='method'>
    <h3>print(String message)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Prints a message to the console.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>executeFile(String filePath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Runs a js script file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>delay(int seconds )</h3>

    <dl>
      <dt>summary</dt>

      <dd>Holds execution a number of seconds</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>clear()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Clears the console.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>say(String message)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Prints a message to the console, adding a trailing
      newline.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>eval(String command)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Evaluates a script.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='kalignws'>kalignws</h2>

  <p class='description'>Controls access to KAlign Webservice at
  EBI.</p>

  <div class='method'>
    <h3>alignDNA(List&lt;IDNA&gt; dnaList)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Align a list of DNA sequences using the KAlign Web
      service at EBI.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>alignProteins(List&lt;IProtein&gt; proteinList)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Align a list of protein sequences using the KAlign Web
      service at EBI.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='matrix'>matrix</h2>

  <p class='description'>Manager to assist working with Bioclipse
  Matrix domain objects.</p>

  <div class='method'>
    <h3>create(String valueSequence, int ncol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new matrix domain object, staring from a
      sequence of values and a given number of columns.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setRowLabels(IMatrixResource matrix, String[] names)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the row labels of the matrix.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addRow(IMatrixResource matrix, List&lt;String&gt;
    values)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a new row to the existing matrix. Throws an
      exception when the number of values is incompatible with the
      number of columns of the existing matrix.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setColumnLabels(IMatrixResource matrix, String[]
    names)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the column labels of the matrix.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>empty()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new, empty matrix domain object.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>create()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new matrix domain object, starting from a array
      of double arrays.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='metaprint2d'>metaprint2d</h2>

  <p class='description'>Contains methods for calling
  MetaPrint2D</p>

  <div class='method'>
    <h3>calculate(List&lt;IMolecule&gt; molecules: List of
    molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run MetaPrint2D on a list of IMolecules. Does not store
      results as properties.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(String path: points to a file comprising one or
    more molecules, storeResults: true if results should be stored
    as properties in AC.)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run MetaPrint2D on a molecular file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run MetaPrint2D on a molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>clear()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Clear MetaPrint2D results for the molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(List&lt;IMolecule&gt; molecules: List of
    molecules, storeResults: true if results should be stored as
    properties in AC.)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run MetaPrint2D on a list of IMolecules.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getActiveDatabase()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Gets the currently active MetaPrint2D database.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(IMolecule molecule, storeResults: true if results
    should be stored as properties in AC.)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run MetaPrint2D on a molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setDataBase(database: String to set; one of "All", "Human",
    "Dog", or "Rat")</h3>

    <dl>
      <dt>summary</dt>

      <dd>Set the active MetaPrint2D Database.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(String path: points to a file comprising one or
    more molecules.)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Run MetaPrint2D on a molecular file.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='myexperiment'>myexperiment</h2>

  <p class='description'>MyExperiment manager.</p>

  <div class='method'>
    <h3>list()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all BSL workflows.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>search(String query)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Search for BSL scripts which have the query string in the
      title.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadWorkflow(Integer workflowNumber, String
    filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download a single workflow from MyExperiment into the
      file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadWorkflow(Integer workflowNumber)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Download a single workflow from MyExperiment.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='opentox'>opentox</h2>

  <p class='description'>Manager that maps the OpenTox API 1.1 to
  manager methods.</p>

  <div class='method'>
    <h3>predictWithModel(String service, String model, IMolecule
    molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Predicts modeled properties for the given molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listAlgorithms(String ontologyServer)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the algorithms available from the given OpenTox
      ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createDataset(String service)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new dataset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>predictWithModelWithLabel(String service, String model,
    IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Predicts modeled properties for the given molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createModel(String algoURI, String datasetURI,
    List&lt;String&gt; featureURIs, String
    predictionFeatureURI)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new regression model using the given algorithm,
      for the given data set, with the features as independent
      variables, and the prediction features as the dependent
      feature.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>searchDescriptors(String ontologyServer, String query)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Search the descriptors available from the given service
      that match the given title search string.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getAlgorithmInfo(String ontologyServer, String
    algorithm)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the available information on the algorithm from the
      given OpenTox ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listDescriptors(String ontologyServer)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the descriptors available from the given OpenTox
      ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadDataSetAsMDLSDfile(String service, String dataSet,
    String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Downloads a data set and saves it as a MDL SD file
      formated file with the given filename.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setDatasetLicense(String datasetURI, String license)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the license of the data set. The license String must
      be a URI.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>login(String user, String password)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Logs in on OpenTox. Returns true if the login
      worked.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateDescriptor(String service, String descriptor,
    IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculates a descriptor value for a single molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listCompounds(String service, Integer dataSet)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the compounds available from the given data
      set.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addMolecules(String datasetURI, List&lt;? extends
    IMolecule&gt; molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a list of molecules to an existing dataset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listFeatures(String service)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the features available from the given service.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listCompounds(String dataSet)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the compounds available from the given data
      set.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getFeatureInfo(String ontologyServer, String feature)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the available information on the feature from the
      given OpenTox ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listDataSets(String service)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the data sets available from the given
      service.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addMolecule(String datasetURI, IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a molecule to an existing dataset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createDataset(String service, IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new dataset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getAlgorithmInfo(String ontologyServer, List&lt;String&gt;
    algorithms)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the available information on the algorithm from the
      given OpenTox ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getModelInfo(String ontologyServer, List&lt;String&gt;
    models)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the available information on the models from the
      given OpenTox ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>search(String service, String inchi)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Searches the OpenTox network for the given molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>search(String service, IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Searches the OpenTox network for the given molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getModelInfo(String ontologyServer, String model)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the available information on the model from the
      given OpenTox ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getToken()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the security token for the service.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>searchModels(String ontologyServer, String query)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Search the models available from the given service that
      match the given title search string.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setDatasetTitle(String datasetURI, String title)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the title of the data set.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>deleteDataset(String datasetURI)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Deletes a dataset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createDataset(String service, List&lt;? extends
    IMolecule&gt; molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new dataset.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>searchDataSets(String ontologyServer, String query)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Search the data sets available from the given service
      that match the given title search string.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculateDescriptor(String service, String descriptor,
    List&lt;? extends IMolecule&gt; molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculates a descriptor value for a set of
      molecules.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadCompoundAsMDLMolfile(String compoundURI)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Downloads a compound and returns it as a MDL molfile
      formated String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listModels(String ontologyServer)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the predictive models available from the given
      service.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>logout()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Logs out on OpenTox.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getFeatureInfo(String ontologyServer, List&lt;String&gt;
    features)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists the available information on the feature from the
      given OpenTox ontology server"s SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>setDatasetRightsHolder(String datasetURI, String
    holder)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sets the rights holder for the data set. The value must
      be a URI.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadCompoundAsMDLMolfile(String service, String
    dataSet, Integer compound)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Downloads a compound and returns it as a MDL molfile
      formated String.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>predictWithModel(String service, String model, List&lt;?
    extends IMolecule&gt; molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Predicts modeled properties for the given list of
      molecules.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>predictWithModelWithLabel(String service, String model,
    List&lt;? extends IMolecule&gt; molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Predicts modeled properties for the given molecule.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='opsin'>opsin</h2>

  <p class='description'>Manager for the OPSIN functionality to
  convert IUPAC names into chemical structures.</p>

  <div class='method'>
    <h3>parseIUPACName(String iupacName)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Converts an IUPAC name into a chemical structure</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>parseIUPACNameAsCML(String iupacName)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Converts an IUPAC name into a CML document</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='owl'>owl</h2>

  <p class='description'>Contains OWL related methods</p>

  <div class='method'>
    <h3>listClasses(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of all OWL classes.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listObjectProperties(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of all OWL object properties.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listDatatypeProperties(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of all OWL data type properties.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='pellet'>pellet</h2>

  <p class='description'>Contains Pellet related methods</p>

  <div class='method'>
    <h3>createStore(String tripleStoreDirectoryPath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new scalable Pellet-targeted store, (using the
      Jena TDB package, which stores on disk as a complement to
      memory, for scalability). tripleStoreDirectoryPath is the
      path (relative to theBioclipse workspace) to a folder to use
      for the triple store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>allAbout(IRDFStore store, String identifier)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns all triples about the Object with the given
      identifier.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isRDFType(IRDFStore store, String type)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the Objects that are of the given rdf:type.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>validate(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Validates the consistency of the RDF model in the
      store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>reason(IRDFStore store, String query)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the results matching the SPARQL query using the
      Pellet reasoner</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createInMemoryStore()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new Pellet-targeted store</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='qsar'>qsar</h2>

  <p class='description'>A manager for QSAR</p>

  <div class='method'>
    <h3>toShortOntologyForm(String ontologyID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Convert a full ontology ID to short ontology ID</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getProviderByID(String providerID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the descriptor provider class by ID</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDescriptorImplsByProvider(String providerID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the ID"s of available descriptors for a
      provider</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(IMolecule mol, String ontologyID, String
    providerID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate a descriptor for a molecule using the default
      provider.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isValidDescriptorDefinition(URL url, String format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Checks if the document behind the URL is a valid QSAR
      descriptor definition. The formats are those accepted by the
      RDF manager: \"RDF/XML\", \"N-TRIPLE\", \"TURTLE\" and
      \"N3\".</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getRealOntologyID(String ontologyID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Convert a short ontology ID to full ID</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDescriptorIDs(boolean hasImpl)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Get a list of descriptor IDs</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>existsDescriptor(String descriptorID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Return true if descriptorID exists</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listDescriptors(boolean hasImpl)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Show all available descriptors. If hasImpl=true, only
      show entries with an available implementation.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDescriptorImpls(String descriptorID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Get a list of descriptor implementations for a
      descriptor</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isValidDescriptorDefinition(String url, String format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Checks if the document behind the URL is a valid QSAR
      descriptor definition. The formats are those accepted by the
      RDF manager: \"RDF/XML\", \"N-TRIPLE\", \"TURTLE\" and
      \"N3\".</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(List&lt;IMolecule&gt; mols, List&lt;String&gt;
    descriptors, String provider )</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate a list of descriptors for a list of molecules
      using the selected provider.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>existsDescriptorImpl(String descriptorID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a descriptor class by ID</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDescriptorIDs()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Get a list of descriptor IDs</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getResponseUnits()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the available response units</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listDescriptors()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Show all available descriptors.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getProviders()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the available descriptor providers</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(IMolecule mol, String ontologyID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate a descriptor for a molecule with the default
      implementation.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getCategoryByID(String categoryID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the descriptor category class by ID</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>show(String ontologyID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Show all implementations for this entry in the
      ontology.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addDescriptors()</h3>
  </div>

  <div class='method'>
    <h3>getDescriptorsInCategory(String categoryID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Get a list of descriptor IDs in a category</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getCategories()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the available descriptor categories</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>calculate(List&lt;IMolecule&gt; mols, List&lt;String&gt;
    descriptors)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Calculate a list of descriptors for a list of molecules
      using the default provider.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDescriptorImplByID(String descriptorImplID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a descriptor class by ID</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='r'>r</h2>

  <p class='description'>Bioclipse manager to interact with R.</p>

  <div class='method'>
    <h3>evalSnippet()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Handles selection and editor content execution</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createMatrix(String varName, IMatrixResource
    matrixData)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new matrix object with the given variablename,
      from the given data.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>ls()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all variables defined in the running R
      environment.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getRightRVersion()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Gets the boolean noting if we have the right R
      version</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>fixFilepath()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the file separator in filepath to /</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>eval(String command)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Evaluates a R command.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='rdf'>rdf</h2>

  <p class='description'>Contains RDF related methods</p>

  <div class='method'>
    <h3>sparql(IRDFStore store, String query)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the results matching the SPARQL query</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>importRDFa(IRDFStore store, String url)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads a XHTML+RDFa file from the URL into the given
      store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addDataProperty(IRDFStore store, String subject, String
    predicate, String value)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a triple to the given store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addPrefix(IRDFStore store, String prefix, String
    namespace)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Registers the prefix for the given namespace.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>importURL(IRDFStore store, String url, Map&lt;String,
    String&gt; extraHeaders, )</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads a RDF/XML file from the URL into the given
      store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>importFile(IRDFStore store, String target, String
    format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads an RDF file in the given content format
      (\"RDF/XML\", \"N-TRIPLE\", \"TURTLE\" and \"N3\") into the
      given store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveRDFNTriple(IRDFStore store, String fileName)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves the store to RDF in N-TRIPLE format.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createInMemoryStore()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new in-memory store.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveRDFN3(IRDFStore store, String fileName)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves the store to a N3 file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>importFromStream(IRDFStore store, InputStream stream,
    String format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads an RDF file in the given content format
      (\"RDF/XML\", \"N-TRIPLE\", \"TURTLE\" and \"N3\") into the
      given store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>allOwlSameAs(IRDFStore store, String resourceURI)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all resources that are owl:sameAs as the given
      resource.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>sparqlConstructRemote(String url, String SPARQL)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Queries a remote SPARQL endpoint and returns RDF/XML.
      Assumes that the query is creating an RDF graph with the
      CONSTRUCT keyword</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>importURL(IRDFStore store, String url)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads a RDF/XML file from the URL into the given
      store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>importFromString(IRDFStore store, String rdfContent, String
    format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads triples from the String in the given format
      (\"RDF/XML\", \"N-TRIPLE\", \"TURTLE\" and \"N3\") into the
      given store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>addObjectProperty(IRDFStore store, String subject, String
    predicate, String object)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Adds a triple to the given store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>asRDFN3(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a String with the N3 serialization.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createStore(String tripleStoreDirectoryPath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new on-disk store, (using the Jena TDB package,
      which stores on disk as a complement to memory, for
      scalability). tripleStoreDirectoryPath is the path (relative
      to theBioclipse workspace) to a folder to use for the triple
      store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>size(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the number of triples in the store</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>allPredicates(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all existing predicates.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>allClasses(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all existing classes.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>saveRDFXML(IRDFStore store, String fileName)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Saves the store to RDF/XML.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>copy(IRDFStore targetStore, IRDFStore sourceStore)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Copies the triples from the sourceStore into the
      targetStore.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getForPredicate(IRDFStore store, String resourceURI, String
    predicate)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all resources or literals for the resource and
      predicate.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>sparqlRemote(String url, String SPARQL)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Queries a remote SPARQL end point.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>dump(IRDFStore store)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Dumps the full model to the returned String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>allOwlEquivalentClass(IRDFStore store, String
    resourceURI)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all resources that are owl:equivalentClass as the
      given resource.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='sadi'>sadi</h2>

  <p class='description'>TODO: Describe the manager here.</p>

  <div class='method'>
    <h3>listServices()</h3>

    <dl>
      <dt>summary</dt>

      <dd>List all registered SADI services</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='signatures'>signatures</h2>

  <p class='description'>Generate Signatures according to Faulon et
  al. 2003,2003, and 2004.</p>

  <div class='method'>
    <h3>generate(String path, int height)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Signatures from SDFile with a given height</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateSparseDataset(List&lt;? extends IMolecule&gt; mols,
    int height, String nameProperty, String responseProperty)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate a sparse dataset of signature counts for a list
      of molecules. Set name of molecules and append response
      values to last column.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate(IMolecule molecule)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Signatures for a molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateDataset(List&lt;? extends IMolecule&gt; mols, int
    height, String nameProperty, String responseProperty)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate a dataset of signature counts for a list of
      molecules. Set name of molecules and append response values
      to last column.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateSparseDataset(List&lt;? extends IMolecule&gt; mols,
    List&lt;Integer&gt; heights, String nameProperty, String
    responseProperty)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate a sparse dataset of signature counts for a list
      of molecules with a list of signatures heights. Set name of
      molecules and append response values to last column.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateMoleculeSignature(IMolecule mol)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate a MolecularSignature from a molecule.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate(List&lt;IMolecule&gt; molecules)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Signatures from a list of molecules</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Signatures from SDFile</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate(IMolecule molecule, int height)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Signatures with a given height for a
      molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generate(List&lt;IMolecule&gt; molecules, int height)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Signatures from a list of molecules with a given
      height</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateChiral(IMolecule molecule, int height)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate Chiral Signatures with a given height for a
      molecule</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateSparseDataset(List&lt;? extends IMolecule&gt; mols,
    int height, )</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate a sparse dataset of signature counts for a list
      of molecules with height h.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>generateDataset(List&lt;? extends IMolecule&gt; mols, int
    height)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Generate a dataset of signature counts for a list of
      molecules with height h.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='twitter'>twitter</h2>

  <p class='description'>Manager that allows interaction with
  Twitter.</p>

  <div class='method'>
    <h3>setStatus()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Sends a status message to the Twitter network</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='ui'>ui</h2>

  <p class='description'>Controls access to Bioclipse UI.</p>

  <div class='method'>
    <h3>isContentType(String type, IContentType contentType)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Checks if the given content type is of the given type,
      either as itself or as one of its base types.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>refresh(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Refresh resource given by path</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>revealAndSelect(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Reveals a file in the Navigator and selects it</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getEditors()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of available editors</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>open(String filePath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens a file in an editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>save(String filePath, String content)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Save the content of the String to the given path.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>openFiles(List&lt;IFile&gt; files)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens a list of files (e.g. a gist) in an editor.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>newFile(String path, String content)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new file at the specified path, with the given
      content.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>newProject(String name)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Create a new project by name.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>assertInstalled(String feature)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Assert a feature is installed.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>readFileIntoArray(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Reads a file line by line into a String[]</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getInstalledFeatures()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of the installed features.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>fileExists(String filePath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether given file path point to a file that
      exists in the workspace</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>open(IBioObject bioObject)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens the IBioObject in its preferred editor</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>closeEditor(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Closes all editors working on a file</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>newFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates an empty file at the specified path</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>closeActiveEditor()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Closes the active editor</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>save(String filePath, InputStream content)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Save the content of the InputStream to the given
      path.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>open(IFile file, String editor)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens a file in an editor specified by ID or alias.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>remove(String filePath)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Deletes a file.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>readFile(String path)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Reads a file line by line into a String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>open(IBioObject bioObject, String editor)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Opens a file in the designated editor, identified by
      editor id or shortname.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='userManager'>userManager</h2>

  <p class='description'>Handles users and accounts in Bioclipse.
  Can store things like database passwords and usernames in an
  encrypted file.</p>

  <div class='method'>
    <h3>reloadFromFile()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Reloads all data from file discarding changes</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getAvailableAccountTypes()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns all available account types</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getLoggedInUsersAccountNames()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the account names of the logged in user</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>deleteUser(String username)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Deletes the user with the given username</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createAccount(String accountID, HashMap&lt;String,
    String&gt; properties, AccountType accountType)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new account of the given accountType and with
      the given properties</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>accountExists(String accountId)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether an account with a given account id
      exists</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getPropertyKeys(String accountId)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the names of the properties for an account with a
      given account id</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getAvailableAccountTypeNames()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the names of all available account types</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>changePassword(String oldPassword, String newPassword)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Changes the password for the logged in user if the
      oldPassword is given correctly</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getUserNames()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns a list of all user names</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>createUser(String username, String password)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a new user with the given username and
      password</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getProperty(String accountId, String property)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the value of the given property for the account
      with the given account id</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getLoggedInUser()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the logged in user</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>persist()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Write all data to file</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>logOut()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Logs out the currently logged in user</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getAccountType(String accountID)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the accounttype for the account with given
      account id</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isLoggedIn()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns whether a user is logged in</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getLoggedInUserName()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the name of the currently logged in user</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>logIn(String username, String password)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Logs in the user with the given username given that the
      given password matches the stored one.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>clearAccounts()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Removes all accounts for the currently logged in
      user</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='ws'>ws</h2>

  <p class='description'>Contains webservices related methods</p>

  <div class='method'>
    <h3>downloadDbEntryAsFile(String db, String query, String
    format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Fetches an entry from the EBI databases and saves it in
      the Webservices result folder</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>queryPDB(String pdbids)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query and retrieve PDB files using theWSDbfetch service
      at EBI.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>queryPDB(String pdbids, String path )</h3>

    <dl>
      <dt>summary</dt>

      <dd>Query and retrieve PDB files using theWSDbfetch service
      at EBI to a specified folder.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadDbEntryAsFile(String db, String query, String
    format, String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Fetches an entry from the EBI databases and saves it
      under the specified filename</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>downloadDbEntry(String db, String query, String
    format)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Fetches an entry from the EBI databases and saves it in
      the Virtual folder</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='xml'>xml</h2>

  <p class='description'>Manager to analyze and validate XML
  documents.</p>

  <div class='method'>
    <h3>validateAgainstXMLSchema(String filename, String
    xmlSchemaFilename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Validates the XML document against the given XML Schema
      document.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>validateAgainstRelaxNG(String filename, String
    relaxngFilename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Validates the XML document against the given RelaxNG
      document.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>validate(String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Validates the XML document against the schemas defined in
      the document itself.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>listNamespaces(String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Lists all namespaces used in the XML document.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>validateAgainstSchematron(String filename, String
    schematronFilename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Validates the XML document against the given Schematron
      1.5 document.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isWellFormed(String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Checks if the XML document is well-formed.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isValid(String filename)</h3>

    <dl>
      <dt>summary</dt>

      <dd>Checks if the XML document is valid against the schemas
      defined in the document itself.</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>

  <h2 id='xmpp'>xmpp</h2>

  <p class='description'>Contains methods for working with
  XWS4J</p>

  <div class='method'>
    <h3>getIoFactory()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Loads an existing IoDataFactory for the specified
      function.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getFunction()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a Function with the specified service JID and
      function name</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>existsIoFactory()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Checks if an IoDataFactory exists for the specified
      function.</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>disconnect()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Disconnect from XMPP server</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getDefaultClient()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the default XMPP Client object</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>isConnected()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns true if connected to XMPP server</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getService()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a Service with the specified JID</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getStatus()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Returns the status of the default XMPP Client</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>connect()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Connect to XMPP server</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>toString()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Converts a DOM element to String</dd>
    </dl>
  </div>

  <div class='method'>
    <h3>getXmppItem()</h3>

    <dl>
      <dt>summary</dt>

      <dd>Creates a XmppItem with the specified JID and node</dd>
    </dl>
  </div>

  <div class='back'>
    <a href='#top'>back to top</a>
  </div>
</body>
</html>
